examples/roll.py

```diff
@@ -0,0 +1,75 @@
+"""An example of a simple CLI for rolling dice.
+
+The example demonstrates the usage of a typed argument
+accepting a list of custom objects (instances of `Die`).
+
+Usage:
+    $ python roll.py 1d6
+    3
+    $ python roll.py 2d6
+    2 5
+    $ python roll.py 1d6 2d3
+    6
+    2 1
+"""
+
+import re
+import random
+from typing import List
+
+from candies.cli import cli, Arg
+
+
+class Die:
```

Comment:
I would expect for `Die` to have only `sides` property, so probably `Roll` would fit better here. So, `1d6` represents a roll to be executed.

Reply:
Yes, maybe. At first, I wanted to use something `die.roll(times)`, and that's where the name comes from. However, I didn't want to rename `Die` to `Roll` simply because `roll(rolls: Arg[List[Roll]])` looks stupid.

Reply:
Is `main` instead of `roll` an appropriate workaround?

Reply:
I like to consider the name of a file also the name of the main command, so that `python roll.py 1d6` looks like calling `roll('1d6')`.